docs/proposals/data-plane-config.md

```diff
@@ -1,21 +1,135 @@
 
-This proposal is intended to contain the design for how to dynamically configure the data plane for the
-NGINX Gateway Fabric (NGF) project. Similar to control plane configuration, we should be able to leverage
+This proposal is intended to contain the design for how to configure global settings for the data plane
+of the NGINX Gateway Fabric (NGF) product. Similar to control plane configuration, we should be able to leverage
 a custom resource definition to define data plane configuration, considering fields such as telemetry and
 upstream zone size.
 
 ## Goals
 
-Define a CRD to dynamically configure various settings for the NGF data plane. The initial configurable options
-will be for telemetry (tracing) and upstream zone size.
+Define a CRD to configure various global settings for the NGF data plane. The initial configurable
+options will be for telemetry (tracing) and upstream zone size.
 
 ## Non-Goals
 
  1. This proposal is not defining every setting that needs to be present in the configuration.
  2. This proposal is not for any configuration related to control plane.
+
+## Introduction
+
+The NGF data plane will evolve to have various user-configurable options. These could include, but are not
+limited to, tracing, logging, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NGF. The first set of options that we will allow users to
+configure are tracing and upstream zone size. The easiest and most intuitive way to implement a Kubernetes-native
+API is through a CRD.
+
+The purpose of this CRD is to contain "global" configuration options for the data plane, and not focused on policy
+per route or backend.
+
+NGF will reload NGINX when configuration changes are made.
+
+In this doc, the term "user" will refer to the cluster operator (the person who installs and manages NGF). The
+cluster operator owns this CRD resource.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxProxy
+metadata:
+    name: nginx-proxy-config
+    namespace: nginx-gateway
+spec:
+    http:
+        upstreamZoneSize: 512k # default
+        telemetry:
+            tracing:
+                enabled: true # default false
+                endpoint: my-otel-collector.svc:4317 # required
+                interval: 5s # default
+                batchSize: 512 # default
+                batchCount: 4 # default
+status:
+    conditions:
+    ...
+```
+
+- The CRD would be Namespace-scoped.
+- CRD is initialized and created when NGF is deployed, in the `nginx-gateway` Namespace.
+- CRD would be referenced in the [ParametersReference][ref] of the NGF GatewayClass.
+- Conditions are either `Valid` or `Invalid` if the configuration was properly set.
```

Comment:
would it make sense to differentiate between Invalid because NGINX failed to reload and invalid because the NginxProxy contents in invalid?

Additionally, I wonder if we should try to mirror Gateway API common conditions like Accepted?

One benefit of it is because NGINX can fail to reload for other reasons - something not related to NginxProxy at all. But in that case, the status will be invalid in both cases. For that case, I was thinking about the programmed condition.